components/Modals/DashboardModal/index.tsx

```diff
@@ -283,15 +284,19 @@ function Component() {
                                     `https://v2.viewblock.io/zilliqa/tx/${deploy[0].ID}?network=${net}`
                                 )
                             }, 1000)
-                            let new_ssi = deploy[0].ContractAddress
-                            new_ssi = zcrypto.toChecksumAddress(new_ssi)
+                            let api = 'https://api.zilliqa.com'
+                            if (net === 'mainnet') {
+                                api = 'https://dev-api.zilliqa.com'
+                            }
+                            const zilliqa = new Zilliqa(api)
+                            const txn =
+                                await zilliqa.blockchain.getContractAddressFromTransactionID(
+                                    deploy[0].ID
+                                )
+                            const new_ssi = '0x' + txn.result
```

Comment:
Remember to always normalise addresses. Also test is buy nft in this case works (have to test from ZilPay mobile @devwistkey @koniandriani)